ts/components/messages/MessageDueDateBar.tsx

```diff
@@ -57,182 +70,210 @@ const styles = StyleSheet.create({
   }
 });
 
-/**
- * A component to show detailed info about the due date of a message
- */
-class MessageDueDateBar extends React.PureComponent<Props> {
-  get paymentExpirationInfo() {
-    return paymentExpirationInfo(this.props.message);
-  }
+// Every possible state of the payment has it's component
+const PaidTextContent: React.FunctionComponent<{
+  onPress: () => void;
+}> = ({ onPress }) => (
+  <>
+    {`${I18n.t("messages.cta.payment.paid")} `}
+    <Text link={true} onPress={onPress}>
+      {I18n.t("wallet.wallet")}
+    </Text>
+  </>
+);
 
-  get paid(): boolean {
-    return this.props.payment !== undefined;
-  }
+const ExpiringTextContent: React.FunctionComponent<{
+  time: string;
+  date: string;
+}> = ({ time, date }) => (
+  <>
+    {I18n.t("messages.cta.payment.expiringAlert.block1")}
+    <Text bold={true} white={true}>{` ${date} `}</Text>
+    {I18n.t("messages.cta.payment.expiringAlert.block2")}
+    <Text bold={true} white={true}>{` ${time} `}</Text>
+  </>
+);
 
-  get isPaymentExpirable(): boolean {
-    return this.paymentExpirationInfo.fold(false, isExpirable);
-  }
+const ExpiredAndExpirableTextContent: React.FunctionComponent<{
+  time: string;
+  date: string;
+}> = ({ time, date }) => (
+  <>
+    {I18n.t("messages.cta.payment.expiredAlert.expirable.block1")}
+    <Text bold={true} white={true}>{` ${time} `}</Text>
+    {I18n.t("messages.cta.payment.expiredAlert.expirable.block2")}
+    <Text bold={true} white={true}>{` ${date}`}</Text>
+  </>
+);
 
-  get isPaymentExpired(): boolean {
-    return this.paymentExpirationInfo.fold(false, isExpired);
-  }
+const ValidTextContent: React.FunctionComponent<{
+  time: string;
+  date: string;
+}> = ({ time, date }) => (
+  <>
+    {I18n.t("messages.cta.payment.addMemo.block1")}
+    <Text bold={true}>{` ${date} `}</Text>
+    {"["}
+    {I18n.t("messages.cta.payment.addMemo.block2")}
+    <Text bold={true}>{` ${time}`}</Text>
+    {"]"}
+  </>
+);
 
-  get isPaymentExpiring(): boolean {
-    return this.paymentExpirationInfo.fold(false, isExpiring);
-  }
+const ExpiredNotExpirableTextContent: React.FunctionComponent = () => (
+  <>{I18n.t("messages.cta.payment.expiredAlert.unexpirable.block")}</>
+);
 
-  get dueDate(): Option<Date> {
-    return fromNullable(this.props.message.content.due_date);
-  }
+const TextContent: React.FunctionComponent<{
+  status: PaymentStatus;
+  dueDate: Date;
+  onPaidPress: () => void;
+}> = ({ status, dueDate, onPaidPress }) => {
+  const time = format(dueDate, "HH.mm");
+  const date = formatDateAsLocal(dueDate, true, true);
+  return status === "paid" ? (
```

Comment:
The ternary operator it's hard to read, often when it is used 2/3 times in the same statement.
Try to use a switch that is more readable and with this, you don't need to use an `assertNever` (if you don't implement all case you'll `have an error not all paths return a value`)

```typescript
switch (status){
    case "paid":
      return <PaidTextContent onPress={onPaidPress} />;
    case "expiring":
      return <ExpiringTextContent time={time} date={date} />;
    case "expiredNotValid":
      return <ExpiredAndExpirableTextContent time={time} date={date} />;
    case "expiredButValid":
      return <ExpiredNotExpirableTextContent />;
    case "valid":
      return <ValidTextContent time={time} date={date} />;

  }
```

Reply:
> if you don't implement all case you'll have an error not all paths return a value

Good point